streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoinTest.java

```diff
@@ -327,17 +341,15 @@ public void testWindowing() {
         assertEquals(1, copartitionGroups.size());
         assertEquals(new HashSet<>(Arrays.asList(topic1, topic2)), copartitionGroups.iterator().next());
 
-        driver.setUp(builder, stateDir);
-
+        driver = new TopologyTestDriver(builder.build(), props, time);
```

Comment:
Yes, using a variable to store the time makes it a bit convoluted. I'll change it to `advanceTimeMs`

Reply:
While I was changing it I realised that, although it would be cleaner to remove `time` and use `advanceTimeMs` instead, it is actually easier to use `time` because the tests require the time to advance and recede.